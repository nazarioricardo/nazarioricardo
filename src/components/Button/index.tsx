import { OverridableComponent } from "@mui/material/OverridableComponent";
import { SvgIconTypeMap } from "@mui/material";
import styles from "./styles.module.css";

type ButtonProps = {
  text: string;
  Icon?: OverridableComponent<SvgIconTypeMap<{}, "svg">> & { muiName: string };
  gradient?: boolean;
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

function Button({ text, Icon, gradient, ...props }: ButtonProps) {
  return (
    <button
      {...props}
      className={`${styles.button} ${gradient && styles.gradient}`}
    >
      {text} {Icon && <Icon />}
    </button>
  );
}

export default Button;
