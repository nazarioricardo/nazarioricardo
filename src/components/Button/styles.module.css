.button {
  display: flex;
  flex-direction: row;
  justify-content: center;
  gap: 12px;
  align-items: center;
  background-color: rgb(var(--background-rgb));
  font-family: var(--font-josefin-sans);
  font-size: 1.1rem;
  height: 52px;
  border: none;
  border-radius: 26px;
  color: rgb(var(--foreground-rgb));
  cursor: pointer;
  text-align: center;
}

.button:hover {
  background-color: rgb(var(--hover-rgb));
}
