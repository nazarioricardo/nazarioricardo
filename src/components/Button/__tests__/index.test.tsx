// Button.test.tsx
import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../index";

describe("Button", () => {
  test("renders Button component", () => {
    render(<Button text="Test Button" />);
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Button text="Test Button" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
