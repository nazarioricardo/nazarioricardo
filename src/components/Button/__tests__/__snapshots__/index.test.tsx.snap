// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button undefined"
  >
    Test Button 
  </button>
</DocumentFragment>
`;
